impl_flock.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+// +build !windows,!linux
+
+package mutex
+
+import (
+	"os"
+	"syscall"
+
+	"github.com/juju/errors"
+)
+
+type impl struct {
+	fd int
+}
+
+func acquire(name string) (Releaser, error) {
+	fd, err := syscall.Open(flockName(name), syscall.O_CREAT|syscall.O_RDONLY, 0600)
```

Comment:
Won't a mode of 0600 mean that only processes belonging to the user which first created the lock will be able to interact with it? That's probably OK for Juju but is a bit surprising for a machine-wide lock.